.gitmodules

```diff
@@ -0,0 +1,9 @@
+[submodule "tools/foundry-example/lib/forge-std"]
```

Comment:
[Forge manages dependencies using git submodules](https://book.getfoundry.sh/projects/dependencies), since the `foundry-example` project does not have git initialised. So when a dependency was installed it likely searched for the nearest parent folder that does i.e. the root directory and so updated this `.gitmodules` file.

Reply:
I think this adds too much overhead to the regular dev. 
Example should just work with a simple git clone and build/run. This adds git submodule considerations that don't make for a good out of box experience.

Can we not add satisfy this with npm packages?
At worst if it's just a few files we need we can pul those over in a directory as the licensing allows.
Some may even suggest an old school Makefile or something to pull in the needed files as part of build.

Thoughts?

Reply:
@Nana-EC hmm, I disagree with:

> I think this adds too much overhead to the regular dev.

as the:

> Example should just work with a simple git clone and build/run.

is in fact how it currently works.

Have you tried checking out to this branch and running it locally? It's really as simple as running `forge install` in the `foundry-example` directory which installs all the dependencies into the `foundry-example/lib/` directory based on the dependencies specified in the `.gitmodules` file

Reply:
Trying it out. Having some issues with the `forge build`.

`Failed to install solc 0.8.20: Unknown version provided`
Debugging to move further with your branch

Reply:
Weighing the options here. This approach brings in a lot of files that are not needed to a devs local machine.
Not all devs will have an appreciation for submodules.

How many files would you need to copy over from the 3 repos into the appropriate folder structures to get it running without submodules. I could only see a few 
The goal of the examples is not to have a large structure that conflates things, rather a simple example that shows how one might start their own project. The approach her is simple but the file footprint seems overly large.
Can we go the the static files approach?

Reply:
@Nana-EC in #1172 I suggested:

> Create mock contracts that mimic the behaviour of Hedera's precompiled contracts and publish them as either:
>> 1. a separate package for easy integration into developers' projects or
>> 2. under a new folder(say hedera-precompiled-mocks) in [hashgraph/hedera-smart-contracts/contracts](https://github.com/hashgraph/hedera-smart-contracts/tree/main/contracts)

I didn't do this as I wanted feedback on which of the 2 options to proceed with.

Once this is done, then it'll be a lot simpler as the `foundry-example` project will only need to install the `hedera-precompiled-mocks` dependency and developers can use the mocks relatively easily without having to concern themselves with all the implementation details, etc.

Reply:
@Nana-EC seems your compilation issue in the build step is related to [foundry-rs/foundry/issues/4933](https://github.com/foundry-rs/foundry/issues/4933)

Maybe try manually setting the [foundry solc version](https://book.getfoundry.sh/reference/config/solidity-compiler#solc_version). In the `foundry.toml` add:
```
solc = "0.8.13"
```

Reply:
A `/mocks` folder under [hashgraph/hedera-smart-contracts/contracts](https://github.com/hashgraph/hedera-smart-contracts/tree/main/contracts) seems like a good idea to do first.
This would have many other developers with mocks to help testing

Reply:
@Nana-EC agreed, I just created [this PR](https://github.com/hashgraph/hedera-smart-contracts/pull/236) for this.